### PR TITLE
Make a new U polygon type.

### PR DESCRIPTION
**Pull request by ikeough**
BACKGROUND:
I wanted a U-shaped polygon as part of our static collection of polygons.

DESCRIPTION:
This PR adds `Polgons.U`.

TESTING:
No new tests have been added. It's just a U.
  
FUTURE WORK:
We should support E, H, and O as other building-shape letter types.

REQUIRED:
- [x] All changes are up to date in `CHANGELOG.md`.

- - -
This change is [Reviewable](https://reviewable.io/reviews/hypar-io/elements/888)

### CODE REVIEW

File: Elements/src/Geometry/Polygons.cs [hunk 1]
```diff
@@ -114,13 +115,45 @@ public static Polygon L(double width, double length, double thickness)
             return new Polygon(true, a, b, c, d, e, f);
         }
 
+        /// <summary>
+        /// Create an U.
```
Comment (wynged) [inline Elements/src/Geometry/Polygons.cs:119]: Create `a` U.

File: Elements/src/Geometry/Polygons.cs [hunk 2]
```diff
@@ -114,13 +115,45 @@ public static Polygon L(double width, double length, double thickness)
...
+        /// </summary>
+        /// <param name="width">The width of the U.</param>
+        /// <param name="length">The length of the U.</param>
+        /// <param name="thickness">The thickness of the U.</param>
+        /// <returns>A U shaped polygon with the origin at the center of
+        /// the inside edge of the U.</returns>
```
Comment (wynged) [inline Elements/src/Geometry/Polygons.cs:125]: I presume this means center of the bottom inside edge?

File: Elements/src/Geometry/Polygons.cs [hunk 3]
```diff
@@ -114,13 +115,45 @@ public static Polygon L(double width, double length, double thickness)
...
+            var a = new Vector3(0, 0, 0);
+            var b = new Vector3(width / 2 - thickness, 0);
+            var c = new Vector3(width / 2 - thickness, length / 2);
+            var d = new Vector3(width / 2, length / 2);
+            var e = new Vector3(width / 2, -length / 2);
+            var f = new Vector3(-width / 2, -length / 2);
+            var g = new Vector3(-width / 2, length / 2);
+            var h = new Vector3(-width / 2 + thickness, length / 2);
+            var i = new Vector3(-width / 2 + thickness, 0);
+            return new Polygon(true, a, b, c, d, e, f, g, h, i);
```
Comment (wynged) [inline Elements/src/Geometry/Polygons.cs:146]: test please, the other shapes are used in tests at least one place it seems.

**Review by wynged — CHANGES_REQUESTED** [review]

**Review by ikeough — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hypar-io/elements/888)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @ikeough and @wynged)

___
*[`Elements/src/Geometry/Polygons.cs` line 119 at r1](https://reviewable.io/reviews/hypar-io/elements/888#-NCMOeYwtGjg4Ph2FZT0-r1-119:-NCMhq8R5goYWEQG4vBZ:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/08d18e9edee4ad94d8c04769121d4cf52831ece3/Elements/src/Geometry/Polygons.cs#L119)):*
<details><summary><i>Previously, wynged (Eric Wassail) wrote…</i></summary><blockquote>

Create `a` U.
</blockquote></details>

Done.
___
*[`Elements/src/Geometry/Polygons.cs` line 146 at r1](https://reviewable.io/reviews/hypar-io/elements/888#-NCMOeYwtGjg4Ph2FZT0-r1-146:-NCMibaQ5nAVhHZmRyoP:b-896fix) ([raw file](https://github.com/hypar-io/elements/blob/08d18e9edee4ad94d8c04769121d4cf52831ece3/Elements/src/Geometry/Polygons.cs#L146)):*
<details><summary><i>Previously, wynged (Eric Wassail) wrote…</i></summary><blockquote>

test please, the other shapes are used in tests at least one place it seems.
</blockquote></details>

Done.

**Review by wynged — APPROVED** [review]

**Pull request merged**